torch/csrc/jit/python/python_sugared_value.cpp

```diff
@@ -219,6 +219,12 @@ Value* ModuleValue::asValue(const SourceRange& loc, Function& m) {
   return self_;
 }
 
+SugaredValuePtr ModuleValue::asTupleValue(const SourceRange& loc, Function& m) {
+  auto dict = getSugaredDict(loc, m);
```

Comment:
Can you error if this isn't a ModuleList here ? nn.ModuleDict doesn't work either (no handling of slices) https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/container.py#L284